controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go

```diff
@@ -60,22 +61,49 @@ func (in *KubeadmControlPlane) ValidateCreate() error {
 	return nil
 }
 
+const (
+	spec                 = "spec"
+	kubeadmConfigSpec    = "kubeadmConfigSpec"
+	clusterConfiguration = "clusterConfiguration"
+)
+
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (in *KubeadmControlPlane) ValidateUpdate(old runtime.Object) error {
+	// add a * to indicate everything beneath is ok.
+	// For example, {"spec", "*"} will allow any path under "spec" to change, such as spec.infrastructureTemplate.name
+	allowedPaths := [][]string{
+		{"metadata", "*"},
+		{spec, kubeadmConfigSpec, clusterConfiguration, "etcd", "local", "imageRepository"},
+		{spec, kubeadmConfigSpec, clusterConfiguration, "etcd", "local", "imageTag"},
+		{spec, "infrastructureTemplate", "name"},
+		{spec, "replicas"},
+	}
```

Comment:
I'm not entirely sure how the status subresource is handled wrt admission webhooks, but do we need to worry about this blocking status updates?

Reply:
Similar to above, maybe we should focus only on spec for now, wydt? I can think that for example in we might want to disallow updates to the cluster-name label in metadata in the future, but we can probably handle that separately

Reply:
I'm not sure either, I'll dig into that and figure it out

Reply:
This is what indicates what will or won't get sent through validation: https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/config/webhook/manifests.yaml#L45-L54

This will only send the spec and none of the sub resources including status so status does not go through validation.

This holds true with the tests that I've run that update the status of the KCP even though status is not a valid field according to the web hook validator.

Reply:
probably also helpful to post the documentation links: https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#webhook-configuration

Reply:
Awesome, thanks for clarifying.